fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAuthenticationConfig.java

```diff
@@ -78,25 +77,36 @@ public FiatService fiatService(FiatClientConfigurationProperties fiatConfigurati
   }
 
   @Bean
-  FilterRegistrationBean fiatFilterRegistrationBean(FiatAuthenticationFilter filter) {
-    val frb = new FilterRegistrationBean(filter);
-    frb.setOrder(Ordered.HIGHEST_PRECEDENCE + 1);
-    frb.addUrlPatterns("/*");
-    return frb;
+  FiatWebSecurityConfigurerAdapter fiatSecurityConfig(@Value("${services.fiat.enabled:false}") boolean isFiatEnabled) {
+    return new FiatWebSecurityConfigurerAdapter(isFiatEnabled);
   }
 
-  @ConditionalOnExpression("!${services.fiat.enabled:false}")
-  @Bean
-  AnonymousConfig anonymousConfig() {
-    return new AnonymousConfig();
-  }
+  private class FiatWebSecurityConfigurerAdapter extends WebSecurityConfigurerAdapter {
+    private final boolean isFiatEnabled;
+
+    private FiatWebSecurityConfigurerAdapter(boolean isFiatEnabled) {
+      this.isFiatEnabled = isFiatEnabled;
+    }
 
-  private class AnonymousConfig extends WebSecurityConfigurerAdapter {
     @Override
     protected void configure(HttpSecurity http) throws Exception {
-      // TODO(ttomsu): Make management endpoints non-sensitive?
-      log.info("Fiat service is disabled. Setting Spring Security to allow all traffic.");
-      http.authorizeRequests().anyRequest().permitAll().and().csrf().disable();
+      if (isFiatEnabled) {
+        /*
+         * Having `FiatAuthenticationFilter` prior to `SecurityContextPersistenceFilter` results in the
+         * `SecurityContextHolder` being overridden with a null value.
+         *
+         * The null value then causes the `AnonymousAuthenticationFilter` to inject an "anonymousUser" which when
+         * passed over the wire to fiat is promptly rejected.
+         *
+         * This behavior is triggered when `management.security.enabled` is `false`.
+         */
+        http
+            .csrf().disable()
+            .addFilterAfter(new FiatAuthenticationFilter(), SecurityContextPersistenceFilter.class);
```

Comment:
Following along through the filter chain, I noticed that `SecurityContextPersistenceFilter` and `AnonymousAuthenticationFilter` basically blew away what the `FiatAuthenticationFilter` was setting in the `SecurityContextHolder`.

It's possible these additional filters came along for the ride when `management.security.enabled` was `false`. 

I'd specifically like to run a service with fiat enabled and the actuator endpoints left open, previously this resulted in a redirect to `/login`.

Would plan on testing with `gate` once https://github.com/spinnaker/gate/pull/553 is merged.

Reply:
I _think_ it works with gate ... at least to the extent that I've tested (gate launches fine and I can see requests go from gate -> fiat with the correct user).

Reply:
Have you tested with auth turned off?

Reply:
In the sense of turning off saml, x509 and ssl ... yes.

No calls to fiat (as expected).